feat: split optimize upgrade to camunda 8 & 7 versions (#4241)
camunda/camunda-docs: optimize/self-managed/optimize-deployment/migration-update/camunda-8/instructions.md

---
id: instructions
title: "Instructions"
description: "Find out how to update to a new version of Optimize without losing your reports and dashboards."
---

Optimize releases two new minor versions a year. These documents guide you through the process of migrating your Optimize from one Optimize minor version to the other.

If you want to update Optimize by several versions, you cannot do that at once, but you need to perform the updates in sequential order. For instance, if you want to update from 3.7 to 3.10, you need to update first from 3.7 to 3.8, then from 3.8 to 3.9, and finally from 3.9 to 3.10. The following table shows the recommended update paths to the latest version:

| Update from        | Recommended update path to 3.6 |
| ------------------ | ------------------------------ |
| 8.6                | You are on the latest version. |
| 3.7 - 3.13.x/8.5.x | Rolling update to 8.6          |

:::note Heads Up!

> Starting with version 8.6, separate artefact are provided for Camunda 7 and Camunda 8. Moving forward, Camunda 8 users should adhere to the 8.x.x versioning format.

Reviewer: ```suggestion
Starting with version 8.6, separate artefacts are provided for Camunda 7 and Camunda 8. Moving forward, Camunda 8 users should adhere to the 8.x.x versioning format.
```

1. Install the new ES version, e.g. using Docker, your favorite package manager, or just by downloading and extracting the new tar/zip archive to a new directory.
2. Copy the data from the old ES to the new ES. If you don't expect any new data coming to your old ES you can just copy the `data` folder from the old ES distribution and overwrite the `data` folder in the new ES distribution.
3. Copy your old configuration (`config/elasticsearch.yml`) over to the new ES installation.
4. Stop the old ES instance.
5. Start the new ES instance and check that everything looks fine.

Although the steps above summarize the basic update procedure, we recommend reading through the Elasticsearch documentation to avoid any potential issues.

:::note Heads Up!

Note that the following updates are not supported by Elasticsearch:

- 6.8 to 7.0.

> - 6.7 to 7.1.–7.X (where X>1, e.g. 7.5)

Reviewer: I feel like we are telling users how to upgrade Elasticsearch, when a link could suffice. Would be interested to know if @akeller thinks we could remove these sections from our migration guide and simply refer to ES docs instead

Reviewer 2: If there is nothing specific to Camunda, we should absolutely link to the ES docs.

Author: I removed the instructions for both Camunda 7 and Camunda 8. It is now updated to the following:

> You only need to execute this step if you want to update the Elasticsearch (ES) version during the update. In case the ES version stays the same, you can skip this step.

> The Elasticsearch update is usually performed in a rolling fashion. Read all about how to do the ES update in the general [Elasticsearch Update Guide](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-upgrade.html) and consult the [rolling ugprade](https://www.elastic.co/guide/en/elasticsearch/reference/current/rolling-upgrades.html) guide of the ES documentation on how to conduct the rolling update.

:::

> ### 3. Perform the migration

Reviewer: maybe we could instead refer to the helm chart docs for migration instructions? Although I think we do also have users not using helm charts for SM 🤔

```
  labels:
    app: optimize
spec:
   initContainers:
     - name: migration
       image: registry.camunda.cloud/optimize-ee/optimize:8-latest
       command: ['./upgrade/upgrade.sh', '--skip-warning']
   containers:
     - name: optimize
       image: registry.camunda.cloud/optimize-ee/optimize:8-latest
```

### 4. Resume a canceled update

Updates are resumable. So if the update process got interrupted either manually or due to an error you don't have to restore the Elasticsearch backup and start over but can simply rerun the update. On resume previously completed update steps will be detected and logged as being skipped. In the following log example **Step 1** was previously completed and is thus skipped:

```
./upgrade/upgrade.sh
...
INFO UpgradeProcedure - Skipping Step 1/2: UpdateIndexStep on index: process-instance as it was found to be previously completed already at: 2020-11-30T16:16:12.358Z.
INFO UpgradeProcedure - Starting step 2/2: UpdateIndexStep on index: decision-instance
...
```

### 5. Typical errors

- Using an update script that does not match your version:

```bash

> Schema version saved in Metadata does not match required [2.X.0]

Reviewer: ```suggestion
Schema version saved in Metadata does not match required [8.X.0]
```
Let's use more current examples to make this clearer

```

> Let's assume have Optimize 2.1 and want to update to 2.3 and use the jar to update from 2.2 to 2.3. This error occurs because the jar expects Elasticsearch to have the schema version 2.1. This is because you downloaded the wrong Optimize artifact which contained the wrong update jar version.

Reviewer: ```suggestion
Let's assume have Optimize 8.4 and want to update to 8.6 and use the jar to update from 8.5 to 8.6. This error occurs because the jar expects Elasticsearch to have the schema version 8.4. This is because you downloaded the wrong Optimize artifact which contained the wrong update jar version.
```

---
id: instructions
title: "Instructions"
description: "Find out how to update to a new version of Optimize without losing your reports and dashboards."
---

Optimize releases two new minor versions a year. These documents guide you through the process of migrating your Optimize from one Optimize minor version to the other.

If you want to update Optimize by several versions, you cannot do that at once, but you need to perform the updates in sequential order. For instance, if you want to update from 3.7 to 3.10, you need to update first from 3.7 to 3.8, then from 3.8 to 3.9, and finally from 3.9 to 3.10. The following table shows the recommended update paths to the latest version:

> | Update from        | Recommended update path to 3.6 |

Reviewer: ```suggestion
| Update from        | Recommended update path to 8.6 |
```

- Start the new Optimize version, as described in the [installation guide](../../install-and-start.md).
- It is very likely that you configured the logging of Optimize to your needs and therefore you adjusted the `environment-logback.xml` in the `config` folder of the root directory of your **old** Optimize. You can now use the backed up logging configuration and put it in the `config` folder of the **new** Optimize to keep your logging adjustments. (Optional)

### 2. Rolling update to the new Elasticsearch version

You only need to execute this step if you want to update the Elasticsearch (ES) version during the update. In case the ES version stays the same, you can skip this step.

The Elasticsearch update is usually performed in a rolling fashion. Read all about how to do the ES update in the general [Elasticsearch Update Guide](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-upgrade.html) and consult the [rolling ugprade](https://www.elastic.co/guide/en/elasticsearch/reference/current/rolling-upgrades.html) guide of the ES documentation on how to conduct the rolling update.

### 3. Perform the migration

> For upgrading the Camunda Helm chart from one release to another, perform a [Helm upgrade](https://docs.camunda.io/docs/self-managed/setup/upgrade/).

Author: I tried using relative urls and the build does not succeed. I tried the following: 
```
[Helm upgrade](/docs/self-managed/setup/upgrade/)
```
I also tried
```
[Helm upgrade](../../../docs/self-managed/setup/upgrade/)
```

> For upgrading the Camunda Helm chart from one release to another, perform a [Helm upgrade](https://docs.camunda.io/docs/self-managed/setup/upgrade/).

Reviewer 3: You'll need to use `$docs$` to get out of the optimize side of things: 
```suggestion
For upgrading the Camunda Helm chart from one release to another, perform a [Helm upgrade]($docs$/self-managed/setup/upgrade).
```
